Fix formations generator

burner1024 opened the pull request:
## Description
make_formations.py is outdated and doesn't match the corresponding 2da's, see #1759.

Hope you don't mind conversion to python 3.
Files are generated with 20 positions.
Files generated with updated script match current 2das exactly, with exception of GATHER formation. I couldn't figure out the algo used to calculate the current one, if there even was any. But first 6 positions match exactly, and the rest shouldn't matter according to [this](https://github.com/gemrb/gemrb/issues/1759#issuecomment-1338761973).

This is the new gather, for reference.
![6-new](https://user-images.githubusercontent.com/1279852/206274041-1b9c2fe9-4a53-4eac-91db-62cd8eb53a8d.png)


## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I used the same coding style as the surrounding code <!-- yet to be formally defined, see issue #161 -->
- [x] I have tested the proposed changes
- [ ] I extended the documentation, if necessary
- [ ] The proposed change builds also on our build bots (check after submission)

lynxlynxlynx commented on admin/make_formation.py:
```diff
@@ -314,6 +315,15 @@ def generate_wedge():
 	print()
 
+def generate_none():
+	print("NONE", end=' ')
+	end = " "
+	for i in range(num_coords):
+		if i == (num_coords - 1):
+			end = ""
+		print("0 0", end=end)
+	print()
```
Why not just use something like `print("NONE ", "0 0 " * (num_coords - 1),  "0 0\n")`

burner1024 replied:
That doesn't generate the same line, but I fixed it.

lynxlynxlynx commented on admin/make_formation.py:
```diff
@@ -77,34 +77,31 @@ def generate_t():
+		elif i == 1:
+			xloc = 48 # right
+			yloc = -24
+		elif i == 2:
+			xloc = -48 # left
+			yloc = -24
```
this block is redundant

burner1024 replied:
All 1-3 are at negative y, and all the rest are at positive.

lynxlynxlynx commented:
Thanks! And yes, a python3 port is welcome. 
It all started with you complaining about formation shape, but these descriptions and code all look like they preserve the existing state. Is the shape fix coming later, did I misunderstand something?

burner1024 commented:
Yes, first I wanted to get the exact state, then make changes.
I think formations are mostly good, except the pentacle one.

lynxlynxlynx commented:
The same files are generated here as well. Please also fix generate_formation_files.sh though, it should be using unhardcoded instead of override. And ideally it would inspect $0 to find out where it is and derive the path prefix from there.

burner1024 commented:
Fixed shell script, and improved pentacle.
Comparison:
![5-2-old](https://user-images.githubusercontent.com/1279852/206321484-0d56e1f0-a1f9-486b-805f-71291644a3d2.png)
![5-new](https://user-images.githubusercontent.com/1279852/206321494-1a4cb815-c498-421f-a692-e8b6b43e760f.png)

bradallred commented:
you should also commit the generated formato.2da files

burner1024 commented:
Converted to draft for now, will revisit the shapes once there's a working projection fix.

burner1024 commented:
Ok, I made Protect and Gather match bg2 more closely, screenshots attached.

Do you want me to review other formations?

![1o](https://user-images.githubusercontent.com/1279852/208002682-3e846eae-f0e6-4969-b422-36f10b09fefe.png)
![1g](https://user-images.githubusercontent.com/1279852/208002656-48ecfe3f-0995-4d10-ac13-9bd1a84c32a5.png)
![2o](https://user-images.githubusercontent.com/1279852/208002703-49792fbe-cc06-4b9b-9d0e-b01625c84a1b.png)
![2g](https://user-images.githubusercontent.com/1279852/208002697-bda6ab45-4420-4471-86e9-c7c4e0ad1a62.png)

lynxlynxlynx commented:
Thanks and please do.

burner1024 commented:
OK, I adjusted the rest, mostly the difference was gemrb variant placing characters closer together than originals.
<details> 
  <summary>Screenshots, originals and updated GemRB</summary>

![wedge_o](https://user-images.githubusercontent.com/1279852/208227671-188f1852-c106-44bb-b255-861a39bf28a6.png)
![wedge_g](https://user-images.githubusercontent.com/1279852/208227673-a7f297fb-1c82-43aa-a2db-08d37766baa6.png)
![tria_o](https://user-images.githubusercontent.com/1279852/208227674-ceb699cb-071e-43d5-b1c3-34b8f0413769.png)
![tria_g](https://user-images.githubusercontent.com/1279852/208227675-85bb2e06-92a9-42fa-b0af-cc9565e9d897.png)
![t_o](https://user-images.githubusercontent.com/1279852/208227676-70e85d16-1966-436f-a382-b5a608d5b89d.png)
![t_g](https://user-images.githubusercontent.com/1279852/208227678-6bccfddc-8622-4ccd-b3c8-af93cdee8309.png)
![s_o](https://user-images.githubusercontent.com/1279852/208227680-dd6a5c8f-2bac-4fbd-b73e-d3bdeb30f76a.png)
![s_g](https://user-images.githubusercontent.com/1279852/208227681-17509269-8a6d-450a-b813-b8fafc1460e2.png)
![rank_o](https://user-images.githubusercontent.com/1279852/208227683-9c64d93a-35c2-4b35-b9d2-a1b945ca42ec.png)
![rank_g](https://user-images.githubusercontent.com/1279852/208227684-0dbc775f-837f-4277-8489-cd866c7f80d8.png)
![line_o](https://user-images.githubusercontent.com/1279852/208227685-702ebc89-ebf2-4a2f-ba0c-57ecc5c12129.png)
![line_g](https://user-images.githubusercontent.com/1279852/208227686-1fd28cd3-7e40-4906-847a-988e83eacdab.png)
![3b2_o](https://user-images.githubusercontent.com/1279852/208227688-6bb04e6e-6a54-4b8c-ae00-ed776157529a.png)
![3b2_g](https://user-images.githubusercontent.com/1279852/208227690-9d01f2ee-a6be-45f4-aff7-d92b1e915c41.png)
![2b3_o](https://user-images.githubusercontent.com/1279852/208227691-afe43ae7-ed09-4f16-9b1c-ed49e3ea66da.png)
![2b3_g](https://user-images.githubusercontent.com/1279852/208227692-cac76b94-7f93-4ec0-a939-15b9829a2e41.png)

</details>

lynxlynxlynx commented:
Thanks!
